dask_ml/feature_extraction/text.py

```diff
@@ -108,3 +112,253 @@ class FeatureHasher(_BaseHasher, sklearn.feature_extraction.text.FeatureHasher):
+            fixed_vocabulary = False
+            # Case 2: learn vocabulary from the data.
+            vocabularies = raw_documents.map_partitions(
+                _build_vocabulary, vocabulary_actor, params
+            )
+            # compute and merge.
+            dask.compute(vocabularies)  # List[Set[str]]
+            # same as before
+            result = raw_documents.map_partitions(
+                _count_vectorizer_transform, vocabulary_actor, params
+            )
+
+        meta = scipy.sparse.eye(0, format="csr", dtype=self.dtype)
+        result = build_array(result, vocabulary_actor.n_features, meta)
+
+        self.vocabulary_actor_ = vocabulary_actor
+        self.vocabulary_ = vocabulary_actor.finalize().result()
+        self.fixed_vocabulary_ = fixed_vocabulary
+
+        return result
+
+    def transform(self, raw_documents):
+        if self.use_actors:
+            return self._transform(raw_documents)
+        else:
+            return self._transform_no_actor(raw_documents)
+
+    def _transform(self, raw_documents):
+        params = self.get_params()
+        del params["vocabulary"]
+        del params["use_actors"]
+
+        check_is_fitted(self, "vocabulary_")
+        result = raw_documents.map_partitions(
+            _count_vectorizer_transform, self.vocabulary_actor_, params
+        )
+        meta = scipy.sparse.eye(0, format="csr", dtype=self.dtype)
+        return build_array(result, self.vocabulary_actor_.n_features, meta)
+
+    def _fit_transform_no_actor(self, raw_documents, y=None):
+        # Just bag for now.
+        # Two cases:
+        # 1. vocabulary provided, easy.
+        # 2. vocabulary learned, harder.
+        params = self.get_params()
+        vocabulary = params.pop("vocabulary")
+        del params["use_actors"]
+
+        if self.vocabulary is not None:
+            # Case 1: Just map transform.
+            fixed_vocabulary = True
+            result = raw_documents.map_partitions(
+                _count_vectorizer_transform_no_actor, dask.delayed(vocabulary), params
+            )
+        else:
+            fixed_vocabulary = False
+            # Case 2: learn vocabulary from the data.
+            vocabularies = raw_documents.map_partitions(
+                _build_vocabulary_no_actor, params
+            )
+
+            # compute
+            vocabulary = _merge_vocabulary(*vocabularies.to_delayed()).compute()
+            try:
+                client = get_client()
+            except ValueError:
+                remote_vocabulary = dask.delayed(vocabulary)
+            else:
+                (remote_vocabulary,) = client.scatter([vocabulary], broadcast=True)
+
+            # same as before
+            result = raw_documents.map_partitions(
+                _count_vectorizer_transform_no_actor, remote_vocabulary, params
+            )
+
+        meta = scipy.sparse.eye(0, format="csr", dtype=self.dtype)
+        result = build_array(result, len(vocabulary), meta)
```

Comment:
we could use `np.nan` rather than `len(vocabulary)` here if we wanted to keep everything lazy.  

All of this might be too much though.  If the vocabulary is small then it probably makes sense to do a separate pass and a full compute call.  I think that what you have here is probably better than what I'm suggesting, except for the `Bag.distinct` call.

Reply:
Going to pass on this. Getting the number of columns right is pretty important / necessary for using this in a pipeline.